src/parser/warrior/arms/modules/core/Overpower.js

```diff
@@ -0,0 +1,70 @@
+import React from 'react';
+
+import { formatPercentage } from 'common/format';
+import SpellLink from 'common/SpellLink';
+import SPELLS from 'common/SPELLS';
+import Analyzer from 'parser/core/Analyzer';
+
+import Enemies from 'parser/shared/modules/Enemies';
+import ExecuteRange from './Execute/ExecuteRange';
+import SpellUsable from '../features/SpellUsable';
+
+class OverpowerAnalyzer extends Analyzer {
+    static dependencies = {
+        executeRange: ExecuteRange,
+        enemies: Enemies,
+        spellUsable: SpellUsable,
+    }
+
+    proc = 0;
+    wastedProc = 0;
+
+    on_byPlayer_cast(event) {
+        if (event.ability.guid !== SPELLS.OVERPOWER.id) {
+            return;
+        }
+
+        this.proc += 1;
+        const overpower = this.selectedCombatant.getBuff(SPELLS.OVERPOWER.id);
+        if (!this.executeRange.isTargetInExecuteRange(event) && !this.selectedCombatant.hasTrait(SPELLS.EXECUTIONERS_PRECISION_TRAIT.id)) {
+            if (overpower !== undefined && overpower.stacks === 2 && this.spellUsable.isAvailable(SPELLS.MORTAL_STRIKE.id)) {
```

Comment:
Checking just for `overpower && overpower.stacks === 2 && this.spellUsable.isAvailable(SPELLS.MORTAL_STRIKE.id)` should be enough, but again, it's not incorrect to explicitly check for `undefined`.

Reply:
If i don't check it for undefined i get a TypeError 
![image](https://user-images.githubusercontent.com/10074100/47726028-fbd46180-dc59-11e8-81b9-9a0ced42d70e.png)

Reply:
That's peculiar (seeing as `executionersPrecision` is not even mentioned here), but we can look into that later.

Reply:
Well... that's probably because i was refering to the wrong file, for the Executioner's Precision trait (in /modules/spells/azeritetraits/) where i also have an undefined check. Sorry it looks like i got lost at some point :/

Reply:
The undefined check is still probably better to leave in, just as a shorthand - `overpower && overpower.stacks ....`, so it doesn't crash if `overpower` is null or undefined (trying to access `stacks` of null or undefined).

Reply:
Can you fix this line as well (few lines above the one you fixed)? It's the same thing, maybe you just forgot 😅 Otherwise good from my side. I asked if someone more experienced could take one last look at entire PR if we haven't missed anything important so we may finally merge it in 😃

Reply:
Yeah sorry missed that one 😅 it's done. Thanks for your reviews !